pkgs/applications/window-managers/river/default.nix

```diff
@@ -51,7 +50,9 @@ stdenv.mkDerivation rec {
 
   installPhase = ''
     runHook preInstall
-    zig build -Drelease-safe -Dcpu=baseline -Dxwayland -Dman-pages --prefix $out install
+    zig build -Drelease-safe -Dcpu=baseline ${
+      lib.optionalString xwaylandSupport "-Dxwayland"
+      } -Dman-pages --prefix $out install
```

Comment:
This is kind of annoying to read as-is IMHO, I would opt for something like this:

```Nix
{
  zigFlags = [
    "-Drelease-safe"
    "-Dcpu=baseline"
    "-Dman-pages"
    "--prefix=${placeholder "out"}"
  ] ++ optional xwaylandSupport "-Dxwayland";

  buildPhase = ''
    runHook preBuild
    zig build ${toString zigFlags}
    runHook postBuild
  '';

  installPhase = ''
    runHook preInstall
    zig install ${toString zigFlags}
    runHook postInstall
  '';
}
```

Double check that separating the build and install steps like I did will work, I'm not familiar with Zig's build system :)

Reply:
I figure something like this should work:
```diff
     export HOME=$TMPDIR
   '';

+  zigFlags = [
+    "-Dcpu=baseline"
+    "-Drelease-safe"
+    "-Dman-pages"
+  ] ++ lib.optional xwaylandSupport "-Dxwayland";
+
   installPhase = ''
     runHook preInstall
-    zig build -Drelease-safe -Dcpu=baseline ${
-      lib.optionalString xwaylandSupport "-Dxwayland"
-      } -Dman-pages --prefix $out install
+    zig build ${toString zigFlags} --prefix $out install
     runHook postInstall
   '';
```
@fortuneteller2k, thoughts?

Reply:
Instead of custom cooking this in every package that uses zig to build something we should introduce this into a hook which zig adds when it is added to nativeBuildInputs. Something for another PR and unrelated.